packages/@aws-cdk/aws-apigatewayv2/lib/http/vpc-link.ts

```diff
@@ -99,10 +99,10 @@ export class VpcLink extends Resource implements IVpcLink {
 
     this.vpcLinkId = cfnResource.ref;
 
-    this.addSubnets(...props.vpc.privateSubnets);
-
-    if (props.subnets) {
```

Comment:
Should we switch `subnets` property from `ec2.ISubnet[]` to `ec2.SubnetSelection`? The latter seems more suitable here.

(Sorry if we've discussed this already in the original PR, and I've forgotten)

Reply:
This is the reasoning I gave in the original PR
>The `vpc` itself only has `ISubnet` array as the member variable and I couldn't find an easy way to convert one to other. Hence kept it `ISubnet`

Let me explore it again once.

Reply:
@nija-at Used `SubnetSelection` instead of `ISubnet[]` and I do agree this is a better API. This will hence be a breaking change. Updated the title description accordingly